### PR TITLE
Simplify Accessors (DRY)

### PR DESCRIPTION
**Pull request by jfparadis**
@warner Just some cleanup for maintainability

### CODE REVIEW

File: shim/src/accessors.js [hunk 1]
```diff
@@ -43,47 +43,29 @@ export function repairAccessors() {
...
+    let base = this;
+    let desc;
+    while (base && !(desc = getOwnPropertyDescriptor(base, prop))) {
+      base = getPrototypeOf(base);
     }
+    return desc && desc.get;
+  }
+
+  defineProperties(Object.prototype, {
+    __defineGetter__: { value: defineAccessor },
```
Comment (ljharb) [inline shim/src/accessors.js]: doesn't this mean that `__defineGetter__ === __defineSetter__`, which isn't supposed to be true?

Comment (jfparadis) [reply shim/src/accessors.js]: Good catch. Let me see.

**Comment by jfparadis** [comment]
Perfomance comparison:
https://jsperf.com/dry-repair-accessors

Difference at most of 5% of 50M ops/sec (5% of 0.000,02ms) on FF

File: shim/src/accessors.js [hunk 2]
```diff
@@ -43,26 +43,20 @@ export function repairAccessors() {
     return `${prop}`;
   }
 
-  defineProperties(Object.prototype, {
-    __defineGetter__: {
+  ['__defineGetter__', '__defineSetter__'].forEach(name =>
+    defineProperty(prototype, name, {
       value(prop, func) {
         return defineProperty(this, prop, {
           get: func,
```
Comment (erights) [inline shim/src/accessors.js]: For the `__defineSetter__` case, this needs to be `set:`, not `get:`

File: shim/src/accessors.js [hunk 3]
```diff
@@ -72,19 +66,8 @@ export function repairAccessors() {
         }
         return desc && desc.get;
```
Comment (erights) [inline shim/src/accessors.js]: For the `__lookupSetter__` case, this needs to be `desc.set`, not `desc.get`.

**Comment by erights** [comment]
The fact that the tests pass anyway indicates what tests we're missing

**Comment by jfparadis** [comment]
Sorry for the breaking submission: I was focusing on code complexity metrics. 
I hear you regarding tests. Stay tuned, I will add them with this PR.

File: shim/src/accessors.js [hunk 4]
```diff
@@ -4,35 +4,43 @@
...
+
+  // On some platforms, the implementation of these functions act as if they are
+  // in sloppy mode: if they're invoked badly, they will expose the global object,
+  // so we need to repair these for security. Thus it is our responsibility to fix
+  // this, and we need to include repairAccessors. E.g. Chrome in 2016.
+
+  function makeDefineAccessor(method, accessor) {
+    defineProperty(prototype, method, {
+      value(prop, func) {
+        defineProperty(this, prop, {
```
Comment (jfparadis) [inline shim/src/accessors.js]: According to specs, we should not return here.

**Comment by jfparadis** [comment]
@erights Tests are in, simply a subset of test262

File: shim/test/test262/annexB/built-ins/Object/prototype/__defineGetter__/define-new.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
```
Comment (jfparadis) [inline shim/test/test262/annexB/built-ins/Object/prototype/__defineGetter__/define-new.js]: I'm still studying the best pattern to bring test262 in. This is useful as is but most likely to evolve.

File: shim/src/accessors.js (hunk 4 above)
Comment (jfparadis) [reply shim/src/accessors.js]: It's supposed to throw (and I added the test):
```
esid: sec-additional-properties-of-the-object.prototype-object
description: Behavior when property exists and is not configurable
info: |
    [...]
    5. Perform ? DefinePropertyOrThrow(O, key, desc).
```

File: shim/src/accessors.js [hunk 6]
```diff
@@ -4,35 +4,43 @@
...
-// import anything from the outside.
-
 // todo: This function is serialized and evaluated outside of the primal
 // realms and it currently can't contain code coverage metrics.
 /* istanbul ignore file */
 export function repairAccessors() {
-  const { getPrototypeOf, defineProperties, defineProperty, getOwnPropertyDescriptor } = Object;
+  const { getPrototypeOf, defineProperty, getOwnPropertyDescriptor, prototype } = Object;
+
+  // On some platforms, the implementation of these functions act as if they are
```
[the remark this answers is not in the document]
Comment (jfparadis) [reply shim/src/accessors.js]: Added comment. FYII: I could not find the detection code in caja or ses.

File: shim/src/accessors.js [hunk 7]
```diff
@@ -4,35 +4,43 @@
...
-// realm's Object.prototype, and we convert it into a string to be executed
-// inside each new RootRealm to fix their Object.prototypes too. So don't
-// import anything from the outside.
-
 // todo: This function is serialized and evaluated outside of the primal
 // realms and it currently can't contain code coverage metrics.
 /* istanbul ignore file */
 export function repairAccessors() {
-  const { getPrototypeOf, defineProperties, defineProperty, getOwnPropertyDescriptor } = Object;
+  const { getPrototypeOf, defineProperty, getOwnPropertyDescriptor, prototype } = Object;
```
[the remark this answers is not in the document]
Comment (jfparadis) [reply shim/src/accessors.js]: Added handling for "false". 
Copied a test262 to check for throw.
Renamed constant.

**Pull request merged**